### PR TITLE
Combine build-info scripts into a single one

### PR DESCRIPTION
**Pull request by kaidokert**
Consolidates all build-info related code into a single target and script.

b/377927950

### CODE REVIEW

File: cobalt/build/build_info.py [hunk 1]
```diff
@@ -143,5 +143,33 @@ def main(output_path, cwd=_FILE_DIR):
...
+  """
+  with open(output_path, 'w', encoding='utf-8') as f:
+    f.write(
+        BUILD_ID_HEADER_TEMPLATE.format(
+            cobalt_build_version_number=get_build_id()))
+
+
 if __name__ == '__main__':
-  main(sys.argv[1])
+  build_id_header(sys.argv[1])
```
Comment (andrewsavage1) [inline cobalt/build/build_info.py:174]: nit: I'd prefer both these functions names to be prefixed with `write_`

Even better would be to have the formatting of the string and the writing be in separate functions as that's more easily testable and the functions would be doing only one thing

Comment (hlwarriner) [reply cobalt/build/build_info.py:174]: +1 to including `write` in the function names if they're going to write.

**Review by hlwarriner — APPROVED** [review]

File: cobalt/build/build_info.py (hunk 1 above)
Comment (kaidokert) [reply cobalt/build/build_info.py]: Added `write_` but i'm not going to do further refactoring here right now, just basic cleanup.

File: cobalt/build/build_info.py [hunk 2]
```diff
@@ -143,5 +143,33 @@ def main(output_path, cwd=_FILE_DIR):
...
+"""
+
+
+def build_id_header(output_path):
+  """Writes a cobalt_build_id header file.
+
+  Args:
+    output_path: Location to write Cobalt build ID header to.
+  Returns:
+    0 on success.
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply cobalt/build/build_info.py:165]: done

[the remark this answers is not in the document]
Comment (kaidokert) [reply cobalt/build/build_info.py]: Removed that, thanks

**Review by oxve — APPROVED** [review]

**Pull request merged**